language-service/src/parser/jsonParser.ts

```diff
@@ -186,11 +186,23 @@ export class ASTNode {
 			if (this.type !== schema.type) {
 				//count strings that look like numbers as strings
 				if (this.type != "number" || schema.type != "string") {
-					validationResult.addProblem({
-						location: { start: this.start, end: this.end },
-						severity: ProblemSeverity.Warning,
-						getMessage: () => schema.errorMessage || localize('typeMismatchWarning', 'Incorrect type. Expected "{0}".', schema.type as string)
-					});
+					let isVariableExpression = false;
+
+					if (this.type === 'string') {
+						const currentValue = String(this.getValue());
```

Comment:
Question - what type does getValue() return? I find the casting to be a bit strange but I also don't have the code in front of me to check at the moment.

Also, can you add a comment in this if-block, saying something like "Ignore expressions as those will be replaced by Azure Pipelines"?

Reply:
Ah, quite annoying...now I remember that this code is _mostly_ typed but with awkward `any`s sprinkled around.